crates/hc_run_local_services/src/lib.rs

```diff
@@ -109,30 +109,62 @@ impl HcRunLocalServices {
     }
 
     pub async fn run_err(self) -> Result<()> {
-        let mut task_list = Vec::new();
+        if self.disable_bootstrap && self.disable_signal {
+            println!("All Services Disabled - Aborting");
+            return Ok(());
+        }
 
+        let (bootstrap_shutdown_tx, bootstrap_shutdown_rx) = tokio::sync::oneshot::channel();
         if !self.disable_bootstrap {
             let bs_ip: std::net::IpAddr = self.bootstrap_interface.parse().map_err(Error::other)?;
             let bs_addr = std::net::SocketAddr::from((bs_ip, self.bootstrap_port));
-            let (bs_driver, bs_addr, shutdown) = kitsune_p2p_bootstrap::run(bs_addr, vec![])
-                .await
-                .map_err(Error::other)?;
-            std::mem::forget(shutdown);
-            task_list.push(bs_driver);
 
-            let mut a_out = AOut::new(&self.bootstrap_address_path).await?;
+            let mut config = kitsune2_bootstrap_srv::Config::testing();
+            config.listen_address_list = vec![bs_addr];
 
-            for addr in tx_addr(bs_addr)? {
-                a_out.write(format!("http://{addr}\n")).await?;
-                println!("# HC BOOTSTRAP - ADDR: http://{addr}");
-            }
+            std::thread::Builder::new()
```

Comment:
The bootstrap server expects to block the main thread, which is being used by Tokio to run the SBD server and listen for ctrl+c so start a thread for the bootstrap server to work from

Reply:
Understood, I noticed the thread handle isn't being joined during shutdown. Would it make sense to store the handle and join it during the shutdown to ensure cleaner termination?

Reply:
Done: https://github.com/holochain/holochain/pull/4741/commits/1e0de733cf6d6015c478afc444b39da0d37c25fe